### ISSUE TITLE
[CLOSED] Paren Issues for Arithmetic Rules.

### ISSUE THREAD

**Issue by eriq-augustine** [linqs/psl#103]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [eriq-augustine](https://github.com/eriq-augustine)**
_Saturday Mar 11, 2017 at 08:40 GMT_
_Originally opened as https://github.com/eriq-augustine/psl/issues/56_

----

```
(0.5 * SameUser(A,B) + 0.5 *  SameUser(B,C)) >= SameUser(A,C)
```
Throws a lex error.

```
0.5 * SameUser(A,B) + 0.5 *  SameUser(B,C) >= SameUser(A,C)
```
Works fine.

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Tuesday Apr 11, 2017 at 05:28 GMT_

----

This is doable, but non-trivial.
It will require changing arithmeticRuleExpression and a decent chunk of terse code:
https://github.com/eriq-augustine/psl/blob/develop/psl-parser/src/main/java/org/linqs/psl/parser/ModelLoader.java#L375

Which is a good thing, I never liked that code.

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Wednesday Apr 12, 2017 at 01:47 GMT_

----

Fixed in 7a87d0c0c7f9fbd8109ff7e2750eab1907f7eea6.
